tests/end-to-end/simple/App.java

```diff
@@ -14,201 +12,137 @@
-
-    bucket.addDependency(queue);
-
-    bucket.applyRemovalPolicy(RemovalPolicy.RETAIN);
-
-    bucketArn = CfnOutput.Builder.create(this, "BucketArn")
-      .value(String.valueOf(Fn.getAtt("Bucket", "Arn")))
-      .description("The ARN of the bucket in this template!")
-      .exportName("ExportName")
-      .condition(isUsEast1)  .build();
-    queueArn = CfnOutput.Builder.create(this, "QueueArn")
-      .value(String.valueOf("Queue"))
-      .description("The ARN of the SQS Queue")  .build();
-    isLarge = CfnOutput.Builder.create(this, "IsLarge")
-      .value(String.valueOf(Fn.conditionIf("IsLargeRegion", true, false)))
-      .description("Whether this is a large region or not")  .build();
-  }
-
-  public static <T> List<String> get(final List<T> input) {
-    return input.stream().map(String::valueOf).collect(Collectors.toList());
-  }
-}
-class GenericList<T> extends LinkedList<T> {
-  public GenericList<T> extend(final T object) {
-    this.addLast(object);
-    return this;
-  }
-
-  public GenericList<T> extend(final List<T> collection) {
-    this.addAll(collection);
-    return this;
-  }
-}
+    private Object isLarge;
 
-class GenericMap<T, S> extends HashMap<T, S> {
-  public GenericMap<T, S> extend(final T key, final S value) {
-    this.put(key, value);
-    return this;
-  }
-
-  public List<CfnTag> getTags() {
-    final List<CfnTag> tags = new LinkedList<>();
-    for (Map.Entry<T, S> entry : this.entrySet()) {
-      tags.add(
-          CfnTag.builder()
-              .key(String.valueOf(entry.getKey()))
-              .value(String.valueOf(entry.getValue()))
-              .build());
+    public Optional<Object> getBucketArn() {
+        return this.bucketArn;
     }
-    return tags;
-  }
-}
 
-class Mapping<T> {
-  private final String name;
-  private final Construct scope;
-  private final Map<String, Map<String, T>> inner = new TreeMap<>();
-
-  public Mapping(Construct scope, String name) {
-    this.name = name;
-    this.scope = scope;
-  }
-
-  public Mapping<T> put(String primaryKey, String secondaryKey, T value) {
-    final Map<String, T> map = inner.getOrDefault(primaryKey, new TreeMap<>());
-    map.put(secondaryKey, value);
-    inner.put(primaryKey, map);
-    return this;
-  }
-
-  public CfnMapping get() {
-    return CfnMapping.Builder.create(this.scope, this.name).mapping(this.inner).build();
-  }
-}
+    public Object getQueueArn() {
+        return this.queueArn;
+    }
 
+    public Object getIsLarge() {
+        return this.isLarge;
+    }
+
+    public SimpleStack(final Construct scope, final String id) {
+        super(scope, id, null);
+    }
+
+    public SimpleStack(final Construct scope, final String id, final StackProps props) {
+        this(scope, id, props, null, null);
+    }
+
+    public SimpleStack(final Construct scope, final String id, final StackProps props,
+            String bucketNamePrefix,
+            String logDestinationBucketName) {
+        super(scope, id, props);
```

Comment:
Not actual feedback I just think the way java forces people to do these multi constructors is really silly

Reply:
I'll let Java know you feel this way the next time we speak.

Reply:
please and thank you 🙏